Bugzilla/WebService/Bugzilla.pm

```diff
@@ -80,6 +85,37 @@ sub time {
     };
 }
 
+sub jobqueue_status {
+    my ( $self, $params ) = @_;
+    
+    Bugzilla->login(LOGIN_REQUIRED);
+
+    my $dbh = Bugzilla->dbh;
+    my $query = q{
+        SELECT
+            COUNT(*) AS total,
+            COALESCE(
+                (SELECT COUNT(*)
+                    FROM ts_error
+                    WHERE ts_error.jobid = j.jobid
+                ) 
+            , 0) AS errors
+        FROM ts_job j
+            INNER JOIN ts_funcmap f
+                ON f.funcid = j.funcid;
+    };
+
+    my $status;
+    try {
+        $status = $dbh->selectrow_hashref($query);
```

Comment:
One more bit: All values that come out of mysql are strings by default, so you need to convince perl that they're numbers.

This should be sufficient:

```perl
        $status->{errors} = 0 + $status->{errors};
        $status->{total}  = 0 + $status->{total};
```

Reply:
Nice catch, tested this and it works great.